app/Http/Transformers/CustomFieldsetsTransformer.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace App\Http\Transformers;
+
+use App\Models\CustomFieldset;
+use App\Models\CustomField;
+use App\Models\AssetModel;
+use Illuminate\Database\Eloquent\Collection;
+use App\Helpers\Helper;
+use Gate;
+
+class CustomFieldsetsTransformer
+{
+
+    public function transformCustomFieldsets (Collection $fieldsets, $total)
+    {
+        $array = array();
+        foreach ($fieldsets as $fieldset) {
+            $array[] = self::transformCustomFieldset($fieldset);
+        }
+        return (new DatatablesTransformer)->transformDatatables($array, $total);
+    }
+
+    public function transformCustomFieldset (CustomFieldset $fieldset)
+    {
+        $fields = $fieldset->fields;
+        $models = $fieldset->models;
```

Comment:
These should probably be eager loaded in CustomFieldsetsController::index(), to avoid n+1.  It doesn't look like they currently are.